File: rest/incoming-phone-numbers/list-post-example-1/list-post-example-1.5.x.php
<?php
// Get the PHP helper library from twilio.com/docs/php/install
require_once '/path/to/vendor/autoload.php'; // Loads the library
use \Twilio\Rest\Client;

// Your Account Sid and Auth Token from twilio.com/user/account
$sid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
$token = "your_auth_token";
$client = new Client($sid, $token);

$number = $client->account->incomingPhoneNumbers
    ->create(
        array(
            "friendlyName" => "My Company Line",
            "voiceUrl" => "http://demo.twilio.com/docs/voice.xml",
            "phoneNumber" => "+15105647903",
            "voiceMethod" => "GET"
        )
    );

echo $number->sid;

File: rest/incoming-phone-numbers/list-post-example-1/list-post-example-1.5.x.rb
# Get twilio-ruby from twilio.com/docs/ruby/install
require 'twilio-ruby'

# Get your Account SID and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'

# Initialize Twilio Client
@client = Twilio::REST::Client.new(account_sid, auth_token)

@number = @client.incoming_phone_numbers
                 .create(
                   friendly_name: 'My Company Line',
                   voice_url: 'http://demo.twilio.com/docs/voice.xml',
                   phone_number: '+15105647903',
                   voice_method: 'GET'
                 )

puts @number.sms_url
